cloudflare/resource_cloudflare_logpush_job.go

```diff
@@ -0,0 +1,176 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCloudflareLogpushJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareLogpushJobCreate,
+		Read:   resourceCloudflareLogpushJobRead,
+		Update: resourceCloudflareLogpushJobUpdate,
+		Delete: resourceCloudflareLogpushJobDelete,
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"logpull_options": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"destination_conf": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ownership_challenge": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func getJobFromResource(d *schema.ResourceData) cloudflare.LogpushJob {
+	id, err := strconv.Atoi(d.Id())
+
+	if err != nil {
+		fmt.Errorf("Could not extract Logpush job from resource - invalid identifier: %+v", id)
+	}
+
+	job := cloudflare.LogpushJob{
+		ID:                 id,
+		Enabled:            d.Get("enabled").(bool),
+		Name:               d.Get("name").(string),
+		LogpullOptions:     d.Get("logpull_options").(string),
+		DestinationConf:    d.Get("destination_conf").(string),
+		OwnershipChallenge: d.Get("ownership_challenge").(string),
+	}
+	return job
+}
+
+func resourceCloudflareLogpushJobRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	jobId, err := strconv.Atoi(d.Id())
+
+	zoneName := d.Get("zone").(string)
```

Comment:
Please skip lookup and use `zone_id` directly.

Reply:
Done in 3fb9208.